cairo_jpg.c

```diff
@@ -91,18 +92,18 @@ cairo_status_t cairo_surface_write_to_jpeg_stream(cairo_surface_t *sfc,
 		cairo_write_func_t write_func, void *closure, int quality) {
 	cairo_status_t e;
 	unsigned char *data = NULL;
-	size_t len = 0;
+	unsigned long len = 0;
 
 	e = cairo_surface_write_to_jpeg_mem(sfc, &data, &len, quality);
-	if (e != CAIRO_STATUS_SUCCESS) {
-		return e;
-	}
+	if (e == CAIRO_STATUS_SUCCESS) {
+		e = len && (len - 1) > (UINT_MAX - 1)
```

Comment:
Why not change `len` to be `unsigned int` instead?

Additionally, can you rewrite this code not to use a ternary operator?

Reply:
Hmm, that's mostly a (corner case) discrepancy of two APIs where
there's a mismatch on type under some circumstances.  In such cases
I'd prefer to keep overflow check in place (now rewritten as an
assertion, is it better?), but for that to be possible, it has to be
deferred under current arrangement, since pass-by-address would drop
that immeditely.  There are many ways to deal with that, primary
objective is not to resort to not using `-Werror`.